fem/bilinearform.hpp

```diff
@@ -292,8 +294,7 @@ class BilinearForm : public Matrix
    /// Returns a const reference to the sparse matrix.
    const SparseMatrix &SpMat() const
    {
-      MFEM_VERIFY(mat, "mat is NULL and can't be dereferenced");
-      return *mat;
+      return const_cast<BilinearForm*>(this)->SpMat();
```

Comment:
Would it also be necessary to add the `` return const_cast<BilinearForm*>(this)->SpMat();`` to the non-const method below? Just for consistency?

Reply:
What this code does is that it uses the non-const version to implement the const version, this is the standard way to implement const and non-const versions of a method (to avoid code duplication). But I can revert this, it's not important any more because I don't change any more the non-const version.

Reply:
Ah okay thanks. I think it is fine if it works for the non-const method as well.